### PR TITLE
Add .claude review tooling, design/security docs, and restructure CLAUDE.md (#409)

### LINKED ISSUE #409: Add repo-specific CLAUDE.md refinements, Claude Code skills/commands, and design + security docs
## Summary / motivation

Give the `coop` repo agent-assisted contribution guardrails: a navigational root `CLAUDE.md`, a `.claude/` directory of custom skills/commands/agents/hooks that enforce quality gates, and dedicated design + security documentation that stays pinned to the code.

coop already had a strong root `CLAUDE.md` (architecture, testing, mutation testing, fuzzing, kani, and a Rust authoring/review checklist), a `SECURITY.md` disclosure policy, and a rich `docs/` tree. What it did **not** have was a `.claude/` directory — no skills, commands, review agents, or hooks — and no standalone architecture or trust-model design doc. This issue adds the missing pieces, adapted to coop's Rust/VM-isolation context.

Why it matters:
- **PR quality guardrails** — automated review lenses catch correctness, convention, doc-drift, and test-coverage problems before a human reviews.
- **Security** — coop stands up a VM isolation boundary and injects credentials into guests; a codified trust model plus a security review agent keep changes from silently weakening that boundary.

### 1. Root `CLAUDE.md` — refine into a navigational entrypoint
Keep the content, but restructure it into a short entrypoint that points at durable docs, moving deep material (mutation-testing scoping, workarounds, Rust idioms) into `docs/`. Add an explicit "Agent Entrypoint" header, a "Trust model" section pointing at the new `docs/trust-model.md`, and a "Pull Requests" section (one PR = one logical change, run gates before opening).

### 2. `.claude/` directory — new
Add, adapted to Rust/cargo and coop's `tests/integration.sh` workflow:

- **Review agents (`.claude/agents/`):** `review-correctness` (Rust logic/edge-case/error-handling — `Result`/`?` propagation, panics, `unwrap` on external input, resource cleanup, SSH/process lifecycle); `review-conventions` (naming/idiom + rename consistency; cross-file sync of `Cargo.toml`, `config.example.toml`, `.pre-commit-config.yaml`, CI workflows, shell installers); `review-design` (simplification, phantom features, one structural finding); `review-docs` (keeps `docs/`, `README.md`, `CLAUDE.md`, `--help`, doc-comments in sync); `review-comments` (inline-comment hygiene); `review-tests` (integration + unit/mutation coverage honoring `.cargo/mutants.toml`); `review-security` (diffs against coop's trust model).
- **Commands (`.claude/commands/`):** `my-review` (build a shared review-context packet and dispatch the agents, post inline findings); `babysit-pr` / `babysit-my-prs` (watch CI, fix failures, address review threads, keep branch current); an integration command wrapping `./tests/run-integration.sh`.
- **Skills (`.claude/skills/`):** `closeout-review` (final gate before commit/push/PR, running the review machinery on the working diff with scope governance); `mutation-check` (guided `cargo mutants` workflow that also verifies `.cargo/mutants.toml` was updated in the same PR — see #352/#373).
- **Hooks + `settings.json`:** `PreToolUse` closeout-review gate before `gh pr create`; `PostToolUse` `cargo fmt` on `Write|Edit`; a curated read-only Bash allowlist to cut permission prompts.

### 3. Design + security docs (`docs/`) — new
- **`docs/ARCHITECTURE.md`** — module map, the two-backend design (Firecracker/Linux vs Lima/macOS), host→guest data flow (setup → up → shell → stop → destroy), architectural invariants.
- **`docs/trust-model.md`** — the authoritative trust boundaries the `review-security` agent reads: the VM as the isolation boundary, credential/secret injection (`secret_store.rs`, `github_pat.rs`, `pat_prompt.rs`), network egress (`network.rs`), guest→host input paths, and `coop update` release/checksum/provenance verification. Complements (not replaces) `SECURITY.md`; cross-linked from both `CLAUDE.md` and `SECURITY.md`.
- **`docs/code-style.md`** — extract the Rust authoring idioms + review checklist currently embedded in `CLAUDE.md`.

## Goals the `.claude/` set must meet

- [ ] PRs are technically and semantically correct — `review-correctness` + `closeout-review`.
- [ ] Coding conventions correct and kept updated — `review-conventions` + `docs/code-style.md`.
- [ ] Design + docs stay in sync with code — `review-docs` + `review-design` + a maintained `docs/ARCHITECTURE.md`.
- [ ] Unit and integration tests reviewed / present — `review-tests`.
- [ ] Security not silently weakened — `review-security` against `docs/trust-model.md`.
- [ ] PR quality/size hygiene — `babysit-pr` + PR conventions in `CLAUDE.md`.

## Open questions

1. **Review dispatch** — CI-triggered review on coop PRs, local-only (`closeout-review`), or both?
2. **Full agent set vs. a subset** — start with the highest-value agents and grow, or port the whole set at once?
3. **CLAUDE.md restructure scope** — restructure now, or add `.claude/` and the new docs first and slim `CLAUDE.md` in a follow-up?
4. **`docs/trust-model.md` vs `SECURITY.md`** — confirm the split: `SECURITY.md` stays the disclosure policy; `trust-model.md` becomes the engineering-facing boundary spec.

### PR DESCRIPTION
**Pull request by evandowning**
Closes #409.

Gives `coop` the same agent-assisted contribution guardrails as `upgrade-do-no-harm`, adapted to coop's Rust/cargo and dual-backend (Firecracker/Lima) workflow: a navigational root `CLAUDE.md`, a `.claude/` directory of review agents/commands/skills/hooks, design + trust-model docs, and an on-demand CI review workflow.

Per the issue's open questions, this PR takes the maximal scope: foundational docs + the full `.claude/` set, restructures `CLAUDE.md` now, and adds both the local review machinery and a CI workflow.

## Docs (`docs/`)

- **`ARCHITECTURE.md`** — module map, the compile-time two-backend design (`VmBackend`/`PlatformBackend`), host→guest data flow, and architectural invariants.
- **`trust-model.md`** — the engineering-facing trust boundaries, taint sources, secret-handling rules, and the `coop update` verification chain. The spec the `review-security` agent reads; `SECURITY.md` stays the disclosure policy and now cross-links it.
- **`code-style.md`** — the Rust authoring idioms and review/authoring checklists (extracted from `CLAUDE.md`).
- **`testing.md`** — integration, mutation testing (full `.cargo/mutants.toml` scoping + baselines), fuzzing, kani.
- **`platform-notes.md`** — Firecracker CI-kernel workarounds, Docker networking trade-off, scp `~` caveat, tracing-to-stderr.
- **`index.md`** — documentation map.

## `CLAUDE.md`

Slimmed from 284 lines into a short navigational entrypoint (agent entrypoint, one-paragraph architecture + trust-model summary, dev commands, before-committing, PR rules). Deep material moved verbatim into the docs above. References that pointed at the relocated content (`CONTRIBUTING.md`, `SECURITY.md`, `docs/json-output-design.md`, and comments in `guest-config.sh`/`lima.rs`/`config.rs`/`paths.rs`) were retargeted.

## `.claude/`

- **`agents/review-*.md`** — eight single-lens PR review sub-agents (correctness, conventions, design, docs, comments, tests, security, api-usage), each scoped to only flag diff-introduced issues.
- **`commands/`** — `my-review`, `babysit-pr`, `babysit-my-prs`, `integration` (wraps `tests/run-integration.sh`).
- **`skills/`** — `closeout-review` (pre-PR gate with scope governance) and `mutation-check` (guided `cargo-mutants` that verifies `.cargo/mutants.toml` was updated in the same PR — the #352/#373 failure mode).
- **`hooks/` + `settings.json`** — `PreToolUse` closeout-review gate before `gh pr create`, `PostToolUse` `cargo fmt`, the no-pipe-test-output guard, and a read-only Bash allowlist.

## CI

`.github/workflows/claude-review.yml` — on-demand review triggered by an `@claude` PR comment, running `/my-review`. **Requires an `ANTHROPIC_API_KEY` repo secret** to function.

## Decisions on the issue's open questions

1. **Review dispatch** — both: the local `closeout-review`/`my-review` machinery and the CI workflow.
2. **Agent set** — the full set of eight.
3. **CLAUDE.md restructure** — done now, in this PR.
4. **Worktree hooks** — omitted. coop has no per-worktree dev-stack setup (unlike DNH's `wt`-based compose stack); the harness's native git worktrees suffice, so DNH-style `WorktreeCreate`/`WorktreeRemove` hooks would be speculative here.
5. **`trust-model.md` vs `SECURITY.md`** — split as proposed: `SECURITY.md` stays the disclosure policy, `docs/trust-model.md` is the engineering boundary spec, cross-linked both ways.

## Verification

- `cargo build` and `cargo fmt -- --check` pass; source edits are comment-only.
- All internal doc links resolve; `settings.json` and the workflow YAML validate; the hook scripts pass `bash -n`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by evandowning** [comment]
@claude review

File: .github/workflows/claude-review.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+jobs:
+  review:
+    # Run only when the comment contains @claude AND is on a pull request
+    # (issue_comment fires for plain issues too; pull_request_review_comment is
+    # always on a PR).
+    if: |
+      contains(github.event.comment.body, '@claude') &&
+      (github.event_name == 'pull_request_review_comment' ||
+       (github.event_name == 'issue_comment' && github.event.issue.pull_request != null))
```
Comment (hbrodin) [inline .github/workflows/claude-review.yml:33]: **No authorization gate — any GitHub user can trigger this and spend the API key.**

The only guard here is `contains(..., '@claude')` + "is a PR". There's no `github.event.comment.author_association` check, so on a public repo *any* user (including a brand-new outside account) can comment `@claude` and kick off a full agent run against the repo's `ANTHROPIC_API_KEY`. The per-PR `concurrency` group only collapses repeats on the *same* PR; commenting across many PRs/issues yields independent runs, and `timeout-minutes` only caps a single run.

The standard mitigation is to gate the `if:` on author association, e.g.:

```yaml
if: |
  contains(github.event.comment.body, '@claude') &&
  contains(fromJSON('["OWNER","MEMBER","COLLABORATOR"]'), github.event.comment.author_association) &&
  (github.event_name == 'pull_request_review_comment' ||
   (github.event_name == 'issue_comment' && github.event.issue.pull_request != null))
```

Worth noting the good news: because the trigger is comment-based, the workflow is read from the trusted base branch and `checkout` takes no PR `ref:`, so untrusted PR-head code is never executed — this is *not* a code-exec pwn-request. The gap is purely authorization. (CI's existing `zizmor` job will likely flag this trigger class too.)

File: .github/workflows/claude-review.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            PR NUMBER: ${{ github.event.issue.number || github.event.pull_request.number }}
+
+            A user requested a review by mentioning @claude in a comment on this pull request.
+
+            Run the /my-review command against this pull request (diff from the PR's base branch).
+            Post substantive findings as inline review comments on the PR, following the
+            "Post findings to the PR" section of the command.
+
+          claude_args: |
+            --allowedTools "Agent,Read,Grep,Glob,WebSearch,WebFetch,mcp__github_inline_comment__create_inline_comment,Bash(git diff:*),Bash(git log:*),Bash(git show:*),Bash(git fetch:*),Bash(gh pr view:*),Bash(gh pr diff:*),Bash(gh pr comment:*),Bash(gh api:*),Bash(cargo tree:*),Bash(cargo metadata:*),Bash(rg:*),Bash(grep:*),Bash(cat:*),Bash(ls:*),Bash(find:*),Bash(jq:*)"
```
Comment (hbrodin) [inline .github/workflows/claude-review.yml]: **Prompt-injection → secret-exfiltration surface in the tool allowlist.**

The agent reads the untrusted PR diff/comment into its context (that's the job), and this allowlist gives it both a way to read secrets and several ways to send them out:

- `Bash(cat:*)` matches `cat /proc/self/environ`, which contains `ANTHROPIC_API_KEY` and `GITHUB_TOKEN` (the Bash tool inherits the process env).
- Exfil channels: `Bash(gh pr comment:*)`, `Bash(gh api:*)`, and **`WebFetch`** (a `WebFetch` to an attacker URL is the cleanest one).
- `Bash(gh api:*)` is also very broad against a `pull-requests: write` token — far more than posting review comments.

A crafted diff/comment could instruct the agent to read the env and post/fetch it out. Combined with the missing author gate above, this is reachable by anyone. Suggest dropping `cat`, `WebFetch`/`WebSearch`, and `gh api` (or narrowing `gh api` to the specific read endpoints `/my-review` needs) — `Read`/`Grep`/`gh pr diff` cover the review's actual needs.

File: .claude/commands/babysit-my-prs.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
+---
+description: Continuously babysit my open PRs — 7-min loop, auto-discovers new PRs, dispatches one background worker per PR needing work.
+allowed-tools: Bash(gh auth status:*), Bash(gh api user:*), Bash(gh repo view:*), Bash(gh search prs:*), Bash(gh pr view:*), Bash(gh api graphql:*), Bash(gh api repos:*), Bash(gh run list:*), Bash(git remote:*), Bash(git rev-parse:*), Bash(git fetch:*), Bash(git worktree:*), Bash(mkdir:*), Bash(ls:*), Bash(stat:*), Bash(date:*), Bash(find:*), Bash(head:*), Bash(echo:*)
```
Comment (hbrodin) [inline .claude/commands/babysit-my-prs.md]: **`allowed-tools` doesn't declare the tools this command actually uses.**

The command's whole mechanism is `CronCreate`/`CronList`/`CronDelete`, `TaskCreate`/`TaskUpdate`, and background `Agent` dispatch, but `allowed-tools` lists only `Bash(...)` entries — none of the orchestration primitives. If `allowed-tools` is treated as the exhaustive allowlist, the command can't create its cron or dispatch workers, i.e. it can't run. (Separately, those `Cron*`/`Task*` primitives may not be present in every harness install — worth confirming they're available where this is expected to run.)

File: .claude/commands/babysit-pr.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
+---
+description: Single-pass PR shepherding — watch CI, fix failures, address review comments, keep the branch up to date with its base.
+allowed-tools: Bash(gh pr view:*), Bash(gh pr diff:*), Bash(gh api graphql:*), Bash(gh api repos:*), Bash(gh run list:*), Bash(gh run view:*), Bash(git fetch:*), Bash(git log:*), Bash(git status:*), Bash(git rev-list:*), Bash(git rev-parse:*), Bash(git diff:*), Bash(git add:*), Bash(git commit:*), Bash(git push:*), Bash(git merge:*), Bash(cargo fmt:*), Bash(cargo clippy:*), Bash(cargo test:*), Bash(cargo check:*), Bash(taplo format:*)
```
Comment (hbrodin) [inline .claude/commands/babysit-pr.md]: **`allowed-tools` is missing the tools the body needs to fix anything.**

Steps 3b/3c call for fixing clippy warnings, fixing the bug behind a failing test, and applying review-comment fixes — all of which need `Read`/`Edit`/`Write` (and `Grep`/`Glob`), none of which are listed. As written it can autofix `cargo fmt`/`taplo format` and commit/push, but stalls on any real code fix. `gh pr comment` (used later in the flow) also isn't in the list, though `gh api graphql`/`gh api repos` are.

**Review by hbrodin — COMMENTED** [review]
Thanks for putting this together, Evan — this is a lot of careful work, and the documentation half in particular is excellent: I spot-checked the new `docs/` tree against the code and every module, type, and trust-model claim resolves, the moved content (mutants.toml scoping, baselines, kani proofs) survived faithfully, and the move even corrected a stale `src/setup.rs` reference. 🙏

Leaving a few inline notes on the **automation** side — the CI workflow and the babysit commands. The `closeout-review` gate being a soft speed-bump is fine by me (human reviewers are the real gate), so I've skipped that; these are the ones I'd want addressed before the automation lands.

File: .github/workflows/claude-review.yml (hunk 1 above)
Comment (evandowning) [reply .github/workflows/claude-review.yml:33]: Fixed in 68f844c: added an author-association gate — the `if:` now also requires `contains(fromJSON('["OWNER","MEMBER","COLLABORATOR"]'), github.event.comment.author_association)`, so drive-by accounts can no longer spend the API key.

File: .github/workflows/claude-review.yml (hunk 2 above)
Comment (evandowning) [reply .github/workflows/claude-review.yml]: Fixed in 68f844c: dropped `cat` (matched `/proc/self/environ`) and `WebSearch`/`WebFetch` from the allowlist, and narrowed `gh api:*` to `gh api graphql:*` (the read-only reviewThreads query `/my-review` needs). File reads now go through `Read`/`Grep`; PR comments remain the only write surface. Also noted in `my-review.md` that the doc-URL-currency check degrades to "unverified" in CI since web tools are withheld.

File: .claude/commands/babysit-my-prs.md (hunk 3 above)
Comment (evandowning) [reply .claude/commands/babysit-my-prs.md]: Fixed in 68f844c: `allowed-tools` now declares `Agent, CronCreate, CronList, CronDelete, TaskCreate, TaskUpdate, TaskList`. Added a Preconditions note that the command depends on those harness primitives and to fall back to `/babysit-pr` where they are not available.

File: .claude/commands/babysit-pr.md (hunk 4 above)
Comment (evandowning) [reply .claude/commands/babysit-pr.md]: Fixed in 68f844c: `allowed-tools` now grants `Read, Edit, Write, Grep, Glob` and `Bash(gh pr comment:*)` so the command can apply real code fixes and post its summary.

**Review by hbrodin — APPROVED** [review]
Approving. I re-reviewed after `68f844c` and it resolves my four inline comments:

- **Author-association gate** (`claude-review.yml`) — added; only `OWNER`/`MEMBER`/`COLLABORATOR` comments can start a run, so a drive-by account can't spend `ANTHROPIC_API_KEY`. This is the load-bearing control.
- **`babysit-my-prs` allowed-tools** — now declares `Agent`, `Cron*`, `Task*`; matches what the body actually orchestrates.
- **`babysit-pr` allowed-tools** — now grants `Read`/`Edit`/`Write`/`Grep`/`Glob` and `gh pr comment`; covers every command the body invokes.
- **Exfil allowlist** — dropping `WebFetch`/`WebSearch` and narrowing `gh api` → `gh api graphql` genuinely helps: exfil is now confined to observable GitHub surfaces, with no silent external channel.

One residual item I'm **not** blocking on: the allowlist can't be made exfil-proof for a review agent (it must read files and post comments), so `grep`/`rg`/`Read` can still read `/proc/self/environ` and a fork PR's diff could carry a prompt-injection that a maintainer triggers via `@claude`. With no outside contributors today this isn't reachable in practice, and the author gate is the real boundary. I've filed #414 to add fork-PR caution + prompt hardening (and to correct the `claude_args` comment, which overstates the allowlist) before external contributions resume.

Good to merge.
